js/controller.js

```diff
@@ -1630,7 +1644,12 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
       OO.log('onRelatedVideosFetched is called');
       if (relatedVideos.videos) {
         this.state.discoveryData = { relatedVideos: relatedVideos.videos };
-        this.state.upNextInfo.upNextData = relatedVideos.videos[0];
+
+        if (relatedVideos.upNextVideo) {
+          this.state.upNextInfo.upNextData = relatedVideos.upNextVideo;
+        } else {
+          this.state.upNextInfo.upNextData = relatedVideos.videos[0];
```

Comment:
This may be null or undefined. Is this ok? Should we also check for relatedVideos.videos.length in the parent if statement?

Reply:
This is the way it's always been, but I'll add a check